shadowsocks/tcp.go

```diff
@@ -61,22 +62,23 @@ func remoteIP(conn net.Conn) net.IP {
 	return nil
 }
 
-func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList) (string, onet.DuplexConn, error) {
+func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList) (string, onet.DuplexConn, int, error) {
 	// This must have enough space to hold the salt + 2 bytes chunk length + AEAD tag (Oeverhead) for any cipher
 	replayBytes := make([]byte, 0, 32+2+16)
 	// Constant of zeroes to use as the start chunk count. This must be as big as the max NonceSize() across all ciphers.
 	zeroCountBuf := make([]byte, 12) // MaxCountSize
 	// To hold the decrypted chunk length.
 	chunkLenBuf := [2]byte{}
 	var err error
+	var maxBytesRead, bytesRead1, bytesRead2 int
```

Comment:
Rather than tracking `maxBytesRead`, I think you can just return `len(replayBytes)`.  However, I don't think you need to do that either, because the number of bytes received is already tracked in `proxyMetrics.ClientProxy`.

Reply:
done